eslint.config.mjs

```diff
@@ -0,0 +1,49 @@
+import eslint from "@eslint/js";
+import eslintConfigPrettier from "eslint-config-prettier/flat";
+import tseslint from "typescript-eslint";
+
+export default tseslint.config(
+    eslint.configs.recommended,
+    tseslint.configs.strictTypeChecked,
+    tseslint.configs.stylisticTypeChecked,
+    eslintConfigPrettier,
```

Comment:
We should probably see how much of this is redundant with the above now.

Reply:
To my knowledge none of these are redundant to each other, but it was a bit ago that I read through those docs.

Reply:
Sorry, My callout was supposed to be on the {} with the custom rule definitions below it.

Reply:
I just went through them, all the custom rule overrides are still valid to https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/eslintrc/recommended-type-checked.ts